### PR TITLE
feature: minimize container image

### PR DESCRIPTION
**Pull request by simar0at**
Rearrange Dockerfile so it creates as little layers as possible and removes everything temporary in the same RUN step as it is downloaded. Like this these downloads are never in any layer and just hidden by deleting them later.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,42 +24,41 @@ ENV NODE_ENV=production
 
 USER root
 
+ADD https://downloads.apache.org/ant/binaries/apache-ant-${ANT_VERSION}-bin.tar.gz \
+    https://github.com/Saxonica/Saxon-HE/releases/download/${SAXON_VERSION}/${SAXON_VERSION}J.zip \
+	https://www.oxygenxml.com/maven/com/oxygenxml/oxygen-patched-xerces/${XERCES_VERSION}/oxygen-patched-xerces-${XERCES_VERSION}.jar \
+	/tmp/
```
Comment (musicEnfanthen) [inline Dockerfile:30]:
```suggestion
ADD https://downloads.apache.org/ant/binaries/apache-ant-${ANT_VERSION}-bin.tar.gz \
    https://github.com/Saxonica/Saxon-HE/releases/download/${SAXON_VERSION}/${SAXON_VERSION}J.zip \
	https://www.oxygenxml.com/maven/com/oxygenxml/oxygen-patched-xerces/${XERCES_VERSION}/oxygen-patched-xerces-${XERCES_VERSION}.jar \
     /tmp/
```

File: Dockerfile [hunk 2]
```diff
@@ -24,42 +24,41 @@ ENV NODE_ENV=production
...
-RUN apt-get update && apt-get install -y --no-install-recommends apt-utils temurin-${JAVA_VERSION}-jdk curl unzip git libc6 aptitude libaom-dev gdebi fonts-stix && \
-    # install prince
-    curl --proto '=https' --tlsv1.2 -O https://www.princexml.com/download/${DEB_FILE} && \
-    gdebi --non-interactive ./${DEB_FILE} && \
+    echo "deb [signed-by=/etc/apt/keyrings/adoptium.asc] https://packages.adoptium.net/artifactory/deb $(awk -F= '/^VERSION_CODENAME/{print$2}' /etc/os-release) main" | tee /etc/apt/sources.list.d/adoptium.list && \
     # install nodejs
     curl -fsSL https://deb.nodesource.com/setup_18.x -o nodesource_setup.sh && \
     bash nodesource_setup.sh && \
-    apt install nodejs && \
+	# install prince
```
Comment (musicEnfanthen) [inline Dockerfile:43]: @simar0at Can you please check indentation here and in the following lines?

**Comment by musicEnfanthen** [comment]
Beside the indentations, looks good to me (we can fix it later if needed. GH UI seems to be contradicting any fixing attempts). Tested locally as well, runs fine.

**Review by musicEnfanthen — APPROVED** [review]

**Comment by musicEnfanthen** [comment]
Will let @bwbohl have another look.

File: Dockerfile (hunk 1 above)
Comment (bwbohl) [reply Dockerfile:30]:
```suggestion
ADD https://downloads.apache.org/ant/binaries/apache-ant-${ANT_VERSION}-bin.tar.gz \
    https://github.com/Saxonica/Saxon-HE/releases/download/${SAXON_VERSION}/${SAXON_VERSION}J.zip \
    https://www.oxygenxml.com/maven/com/oxygenxml/oxygen-patched-xerces/${XERCES_VERSION}/oxygen-patched-xerces-${XERCES_VERSION}.jar \
    /tmp/
```

also applicable to line above

Comment (musicEnfanthen) [reply Dockerfile:30]: I guess we can fix this later on as well. I recognized that the corrections in the github ui do not correctly set the indentations.

**Comment by bwbohl** [comment]
I created a PR with the fixed against @simar0at ’s
Many thanks; that really compresses the image by about 300MB!

I was wondering whether we should also move the download of ANT, Saxon, and Xerces (line 27) into the RUN (line 35). Doing so, we would drop another layer, and this should further reduce image size, correct?

**Review by bwbohl — CHANGES_REQUESTED** [review]
created a PR with the corresponding indentation

**Comment by simar0at** [comment]
Downloading in a big `RUN` might reduce the size a little but you will get the biggest reduction if you then also delete install packages you downloaded in that `RUN` statement.

**Comment by bwbohl** [comment]
Deleting the zips after deploying them would be absolutely fine

File: Dockerfile (hunk 2 above)
Comment (bwbohl) [reply Dockerfile]: just realised that the `apt install nodejs && \` was dropped, can’t recall if or why this was necessary… ideas?

**Comment by simar0at** [comment]
```
145.2 E: Failed to fetch https://packages.adoptium.net/artifactory/deb/pool/main/t/temurin-17/temurin-17-jdk_17.0.10.0.0%2b7_arm64.deb  Error reading from server. Remote end closed connection [IP: 146.75.39.42 443]
```

Either the logic for getting temurin is broken or this is a one off and just rerunning will du.

**Comment by musicEnfanthen** [comment]
Rerun was fine

**Review by bwbohl — APPROVED** [review]

**Comment by musicEnfanthen** [comment]
Thank you @simar0at for this helpful contribution!

**Pull request merged**